consensus/misc/eip1559.go

```diff
@@ -51,8 +51,11 @@ func VerifyEip1559Header(config *params.ChainConfig, parent, header *types.Heade
 
 // CalcBaseFee calculates the basefee of the header.
 func CalcBaseFee(config *params.ChainConfig, parent *types.Header, parentL1BaseFee *big.Int) *big.Int {
-	l2SequencerFee := big.NewInt(1000000) // 0.001 Gwei
-	provingFee := big.NewInt(47700000)    // 0.0477 Gwei
+	if config.Clique != nil && config.Clique.ShadowForkHeight != 0 && parent.Number.Uint64() >= config.Clique.ShadowForkHeight {
```

Comment:
```suggestion
	if config.Clique != nil && config.Clique.ShadowForkHeight != 0 && parent.Number.Uint64() + 1 >= config.Clique.ShadowForkHeight {
```

Reply:
isn't it current height >= shadowfork height, or it's parent height >= shadowfork height?

Reply:
> it's parent height >= shadowfork height

this. 

`number` [here](https://github.com/scroll-tech/go-ethereum/pull/828/files#diff-1f9c721fa9815697c0778c180de0f86da1f7c16297d786412e015700a3575362R380) is the parent's height. `snapshot()` is always called with `currentHeight-1` ie parent's height.

Reply:
oh understand and thanks. a followup question: I think it's a discussion of the begin height of shadowfork, so is there a canonical practice (like definition) of shadowfork height? if so we can follow it.

Reply:
I don't think there is, we just need to be consistent within our codebase.

Reply:
got it.